### PR TITLE
fix(lint): Change endOfLine rules to better support linting on Windows

### PR DESCRIPTION
**Pull request by HowardBraham**
Before this PR, if you run eslint on Windows, you will get thousands of CRLF errors like this:
```
    1:52   error  Delete `␍`  prettier/prettier
    2:52   error  Delete `␍`  prettier/prettier
    3:74   error  Delete `␍`  prettier/prettier
```
You can fix them with `yarn lint:fix`, but they will probably come back again, depending on your git and text editor settings.

Meanwhile, git commits everything as LF, so it shouldn't affect Mac/Linux.

**Notes:**
- Made similar changes to eslint-config, but metamask-extension is currently on an old version of eslint-config, so putting it in this repo as well
MetaMask/eslint-config#311
- This line in package.json was invalid syntax: `"lint:styles": "stylelint -- */**/*.scss",`
and on Windows, produced the error `The command line is too long.`
Changed to the valid `"lint:styles": "stylelint '*/**/*.scss'",`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -63,7 +63,7 @@
     "lint:lockfile:dedupe:fix": "yarn dedupe",
     "lint:lockfile": "lockfile-lint --path yarn.lock --allowed-hosts npm yarn github.com codeload.github.com --empty-hostname true --allowed-schemes \"https:\" \"git+https:\" \"npm:\" \"patch:\" \"workspace:\"",
     "lint:shellcheck": "./development/shellcheck.sh",
-    "lint:styles": "stylelint -- */**/*.scss",
```
Comment (mcmire) [inline package.json:66]: Is this change necessary because lint wasn't running at all on Windows?

Comment (HowardBraham) [reply package.json:66]: See the second bullet point in the PR description: "on Windows, produced the error `The command line is too long.`"

**Review by mcmire — APPROVED** [review]
Looks good to me.

**Review by NidhiKJha — APPROVED** [review]
